log/logger.go

```diff
@@ -76,8 +76,15 @@ func NewLoggerWithField(key string, value interface{}) Logger {
 	if logger == nil {
 		InitLogger("debug", "") // nolint: errcheck, gosec	// err will always be nil in this case.
 	}
-	l := logger.WithField(key, value)
-	return l
+	return logger.WithField(key, value)
+}
+
+// NewDerivedLoggerWithField returns a logger with the fields logged along with the parentlogger.
+func NewDerivedLoggerWithField(parentLogger Logger, key string, value interface{}) Logger {
+	if parentLogger == nil {
+		InitLogger("debug", "") // nolint: errcheck, gosec	// err will always be nil in this case.
+	}
```

Comment:
```suggestion
	if parentLogger == nil {
		panic("parent logger should not be nil").
	}
```
	
These lines have no impact, the function will still panic if parent logger is nil. Hence replacing `InitLogger` with a `panic`. I have added this info to doc comments of the function as well.